07-documentation.md

```diff
@@ -0,0 +1,80 @@
+	public .... createSomething(..)
+
+Swagger is easy to integrate in projects but also has various drawbacks:
+
+- *No cross cutting concerns.* Swagger only shows the requests, there is no way to define cross cutting concerns. One clear example is the usage of error codes. Each request shows a list of all possible error codes, most are clearly irrelevant for that request, but could confuse a frontend developer. The only way to overcome this problem is by heavily annotating your controller with Swagger annotations.
+- *Grouping per controller not resource.* Sometimes a resource is split over various controllers for design reasons. Each controller is shown seperately, while you would rather see them as one resource.
+
+## Spring REST docs
+
+[Spring REST docs](http://docs.spring.io/spring-restdocs/docs/1.0.x/reference/html5/) helps to generate an acurate and readable REST document. Rather than a testsuite we will produce an actual document, using the text processor [Ascii doctor](http://asciidoctor.org/). In this document we can manually describe the cross cutting concerns and resources, allowing us to manage our own markup and chapters.
+
+Frequently changing information, such as HTTP request details, are generated. Spring REST docs generates various snippets from unit tests written in [Spring MVC Test](http://docs.spring.io/spring/docs/current/spring-framework-reference/htmlsingle/#spring-mvc-test-framework). By using unit tests you ensure that the documentation is actual, otherwise the test will fail. Another nice side effect is that each request will be tested.
+
+Ascii doctor allows us to include the generated snippets in our document, with synthax highlighting. This way the document is always acurate, while maintaining control over the document.
+
+## Practice
+
+Both tools are actively used in projects, but Spring REST docs is preferred. There is a clear difference between both tools. Swagger guesses how your controller should be used, while Spring REST docs allows you to demonstrate the usage in a unit test. Spring REST docs will result in a much clearer and relevant document, staying acurate during code changes.
+
+# Document
+
+When writing a REST document ensure the following subjects are discussed:
+- Cross cutting concerns
+- Resources: Introduction, Requests, Relation with other resources
+
+We also recommend referencing to this REST convention, allowing various trivial subjects to be skipped.
+
+## Example resource
+
+To make this clearer we will provide an example resource description below.
+
+### User
+
+Represents a person that uses our system. Each user has a role, which is also a seperate resource.
+
+- Get a user
+- Create user
+- Update user
+- Remove user
+
+**Get a user**
+
+	GET /api/users/1
+
+
+	{
+		id: 1,
+		email: developer@42.nl,
+		role: {
+			name: "ADMIN"
+		}
+	}
+
+Another nice example is [Github](https://developer.github.com/v3/git/commits/). Where they describe their *Commit* resource.
```

Comment:
Maybe change it in something more like: `A good example of a REST documentation is [this](https://developer.github.com/v3/git/commits/)  where GitHub describes their *commit* resource of their REST api.`

And maybe place under a separate header? Now it looks like it is part of the user example resource.